backend/kernelCI_app/views/treeView.py

```diff
@@ -89,25 +73,58 @@ def get(self, request):
                     AND tests.status = 'DONE' THEN 1 END) AS done_boots,
                 COUNT(CASE WHEN (tests.path = 'boot' OR tests.path LIKE 'boot.%%')
                     AND tests.status = 'SKIP' THEN 1 END) AS skip_boots,
-
                 COALESCE(
-                    ARRAY_AGG(DISTINCT tree_name) FILTER (WHERE tree_name IS NOT NULL),
-                    ARRAY[]::text[]
+                    ARRAY_AGG(DISTINCT tree_name) FILTER (
+                        WHERE tree_name IS NOT NULL
+                    ),
+                    ARRAY[]::TEXT[]
                 ) AS tree_names
             FROM
-                selection
-            JOIN checkouts ON
-                checkouts.git_repository_url = selection.git_repository_url AND
-                checkouts.git_repository_branch = selection.git_repository_branch AND
-                checkouts.start_time = selection.start_time
+                checkouts
             LEFT JOIN
                 builds ON builds.checkout_id = checkouts.id
             LEFT JOIN
                 tests ON tests.build_id = builds.id
+            WHERE
+                checkouts.git_commit_hash IN (
+                    SELECT
+                        git_commit_hash
+                    FROM
+                        (
+                            SELECT
+                                id,
+                                tree_name,
+                                git_repository_branch,
+                                git_repository_url,
+                                git_commit_hash,
+                                git_commit_name,
+                                patchset_hash,
+                                start_time,
```

Comment:
do you need to select all these things for this subquery? `ordered_checkouts_by_tree` doesn't seem to be used anywhere else

Reply:
Some of then were indeed not necessary (as far as I tested) so I removed them